src/edu/stanford/nlp/tagger/maxent/TestSentence.java

```diff
@@ -574,60 +567,39 @@ private void cleanUpScorer() {
     return scores;
   }
 
-  // todo [cdm 2016]: Could this be sped up a bit by caching lambda array, extracting method for shared code?
-  // todo [cdm 2016]: Also it's allocating java.util.ArrayList$Itr for for loop - why can't it just random access array?
   /** Returns an unnormalized score (in log space) for each tag. */
   private double[] getApproximateHistories(String[] tags, History h, List<Pair<Integer,Extractor>> extractors, List<Pair<Integer,Extractor>> extractorsRare) {
-
     double[] scores = new double[tags.length];
-    int szCommon = maxentTagger.extractors.size();
     if (VERBOSE) { System.err.println("Calling approx histories"); }
+    double[] lambda = maxentTagger.getLambdaSolve().lambda;
 
-    for (Pair<Integer,Extractor> e : extractors) {
-      int kf = e.first();
-      Extractor ex = e.second();
-      String val = ex.extract(h);
-      int[] fAssociations = maxentTagger.fAssociations.get(kf).get(val);
-      if (fAssociations != null) {
-        for (int j = 0; j < tags.length; j++) {
-          String tag = tags[j];
-          int tagIndex = maxentTagger.tags.getIndex(tag);
-          int fNum = fAssociations[tagIndex];
-          if (fNum > -1) {
-            double score = maxentTagger.getLambdaSolve().lambda[fNum];
-            if (VERBOSE) {
-              fullScores[kf][j] = score;
```

Comment:
Can we keep the debug statements?  They didn't make it into the refactored method below.

Reply:
They were added to the codebase after my initial version, which is how they got lost. Looked like dead code when rebasing. I'll add them back.